index.js

```diff
@@ -835,6 +871,68 @@ class Batch {
   }
 }
 
+class Watcher extends EventEmitter {
+  constructor (bee, range) {
+    super()
+
+    this.bee = bee
+    this.core = bee.core
+
+    this.running = false
+    this.destroyed = false
+
+    this.range = range
+    this.latestDiff = this.bee.version
```

Comment:
you should set this in _run for the first run, ie the "ready" step. otherwise the bee has to be ready'ed first, which should not be a requirement

Reply:
ie

```js
_run () {
  if (first) {
    await this.bee.ready()
    this.latestDiff = this.bee.version
    return
  }
  ...
}
```

And then trigger the debouncer once from the constructor manually with a safetyCatch .catch

Reply:
I was thinking about the ready step at the beginning then I did this this test to prove that is not needed:
https://github.com/holepunchto/hyperbee/blob/ee5d3f409f47d127a410cb2599b9b97859f2d25d/test/watch.js#L87
But you're right, and also just realized the test is wrong

Reply:
Okey, this is the new test that confirms that it should be working good now:
https://github.com/holepunchto/hyperbee/blob/1bec4e71bfa373623afc3758cb12bc769ffe73fe/test/watch.js#L130
`oldVersion` was resulting in `1` before, but now `3` which sounds correct